components/MultipleFilesMenu.vue

```diff
@@ -0,0 +1,155 @@
+<script setup lang="ts">
+import { FileObject } from "@supabase/storage-js";
+
+const selectedFiles = useSelectedFiles();
+const deleteFile = useDeleteFile();
+const downloadFile = useDownloadFile();
+const copyFile = useCopyFile();
+
+const props = defineProps<{
+  fileList: FileObject[];
+}>();
+
+const emit = defineEmits<{
+  (e: "filesAction"): void;
+}>();
+
+const showErrorModal = ref(false);
+const errorMessages = ref<string[]>([]);
+
+const openModal = () => {
+  if (errorMessages.value.length > 0) {
+    showErrorModal.value = true;
+  }
+};
+
+const closeModal = () => {
+  showErrorModal.value = false;
+  errorMessages.value = [];
+};
+
+watch(errorMessages, openModal);
+
+const numberOfSelectedFiles = computed(() => {
+  return selectedFiles.value.length;
+});
+
+const computedWrapperClass = computed(() => {
+  return numberOfSelectedFiles.value === 0 ? "inactive" : "";
+});
+
+const isDisabled = computed(() => {
+  return numberOfSelectedFiles.value === 0;
+});
+
+const numbOfSelectedFilesLabel = computed(() => {
+  return numberOfSelectedFiles.value === 1
+    ? `${numberOfSelectedFiles.value} file selected`
+    : `${numberOfSelectedFiles.value} files selected`;
+});
+
+const handleClearSelection = () => {
+  selectedFiles.value = [];
+};
+
+const handleCopyFiles = async () => {
+  try {
+    await Promise.all(
+      selectedFiles.value.map((file) => {
+        return copyFile.copy(file, props.fileList);
+      })
+    );
+  } catch (error: any) {
+    errorMessages.value.push(error.message);
+  }
+  emit("filesAction");
+  handleClearSelection();
+};
+
+const handleDownloadFiles = () => {
+  selectedFiles.value.forEach((file, index) => {
+    const download = () => {
+      downloadFile.download(file);
+    };
+    setTimeout(download, Number(`${index}000`));
+  });
+  handleClearSelection();
+};
```

Comment:
Downloading file is performed by creating an anchor tag in composable, clicking on it - then file is downloaded. It might be browser behaviour that there is no option to download separate multiple files at once, so here I used setTimeout to enable downloading files one after another. My tests showed that if I used shorter time eg: Number(`${index}00`) it still doesn't work, that's why I set Number(`${index}000`).